src/cipher/encoder/encoder.go

```diff
@@ -466,6 +502,18 @@ func datasizeWrite(v reflect.Value) (int, error) {
 	}
 }
 
+func parseTag(tag string) (string, bool) {
+	tagSplit := strings.Split(tag, ",")
+	name := tagSplit[0]
+
+	omitempty := false
+	if len(tagSplit) > 1 && tagSplit[1] == "omitempty" {
```

Comment:
Trim space and use `tagSplit[len(tagSplit)-1]` to get the last tag might be better.

Reply:
The tag format follows json package